modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/position/align.js

```diff
@@ -0,0 +1,240 @@
+			left =
+				targetRectangle.left +
+				targetRectangle.width / 2 -
+				sourceRectangle.width / 2;
+			break;
+		case Align.RightCenter:
+			top =
+				targetRectangle.top +
+				targetRectangle.height / 2 -
+				sourceRectangle.height / 2;
+			left = targetRectangle.left + targetRectangle.width;
+			break;
+		case Align.BottomCenter:
+			top = targetRectangle.bottom;
+			left =
+				targetRectangle.left +
+				targetRectangle.width / 2 -
+				sourceRectangle.width / 2;
+			break;
+		case Align.LeftCenter:
+			top =
+				targetRectangle.top +
+				targetRectangle.height / 2 -
+				sourceRectangle.height / 2;
+			left = targetRectangle.left - sourceRectangle.width;
+			break;
+		case Align.TopRight:
+			top = targetRectangle.top - sourceRectangle.height;
+			left = targetRectangle.right - sourceRectangle.width;
+			break;
+		case Align.BottomRight:
+			top = targetRectangle.bottom;
+			left = targetRectangle.right - sourceRectangle.width;
+			break;
+		case Align.BottomLeft:
+			top = targetRectangle.bottom;
+			left = targetRectangle.left;
+			break;
+		case Align.TopLeft:
+			top = targetRectangle.top - sourceRectangle.height;
+			left = targetRectangle.left;
+			break;
+		default:
+			break;
+	}
+
+	return {
+		bottom: top + sourceRectangle.height,
+		height: sourceRectangle.height,
+		left,
+		right: left + sourceRectangle.width,
+		top,
+		width: sourceRectangle.width,
+	};
+}
+
+/**
+ * Looks for the best region for aligning the given element. The best
+ * region is defined by clockwise rotation starting from the specified
+ * `position`. The element is always aligned in the middle of alignElement
+ * axis.
+ * @param {!Element} element Element to be aligned.
+ * @param {!Element} alignElement Element to align with.
+ * @param {Align.Top|Align.Right|Align.Bottom|Align.Left} position
+ *     The initial position to try. Options `Align.Top`, `Align.Right`,
+ *     `Align.Bottom`, `Align.Left`.
+ * @return {{position: string, region: DOMRect}} Best region to align element.
+ */
+function suggestAlignBestRegion(element, alignElement, position) {
+	let bestArea = 0;
+	let bestPosition = position;
+	let bestRegion = getAlignRegion(element, alignElement, bestPosition);
+	let tryPosition = bestPosition;
+	let tryRegion = bestRegion;
+	const viewportRegion = getRegion(window);
+
+	for (let i = 0; i < 8; ) {
+		if (intersectRegion(viewportRegion, tryRegion)) {
+			const visibleRegion = intersection(viewportRegion, tryRegion);
+			const area = visibleRegion.width * visibleRegion.height;
+			if (area > bestArea) {
+				bestArea = area;
+				bestRegion = tryRegion;
+				bestPosition = tryPosition;
+			}
+			if (insideViewport(tryRegion)) {
+				break;
+			}
+		}
+		tryPosition = (position + ++i) % 8;
+		tryRegion = getAlignRegion(element, alignElement, tryPosition);
+	}
+
+	return {
+		position: bestPosition,
+		region: bestRegion,
+	};
+}
+
+export {
+	Align,
+	align,
+	getAlignBestRegion,
+	getAlignRegion,
+	suggestAlignBestRegion,
+};
```

Comment:
Independently of the cool discussion about `Align` vs `align`, these should probably be exported as you go, instead of all at once at the end (ditto in position.js too).

Reply:
OK, I'm inlining the contents of `position.js` into `align.js` but I'll export as I go, thanks.